internal/confmapprovider/discovery/discoverer.go

```diff
@@ -114,7 +109,6 @@ func newDiscoverer(logger *zap.Logger) (*discoverer, error) {
 		configs:                   map[string]*Config{},
 		duration:                  duration,
 		mu:                        sync.Mutex{},
-		expandConverter:           expandconverter.NewFactory().Create(confmap.ConverterSettings{}),
```

Comment:
Lint is failing as `expandConverter` is no longer used, looks like we can remove it from the `discoverer` struct.

Reply:
isn't discoverer used by the discovery mechanism somewhere?